### PR TITLE
Show page

### PR DESCRIPTION
**Pull request by Rileyp33**
- Refactored index page to use React
- Added show page in React and related serializers
- Enzyme tests for all React components

### CODE REVIEW

File: db/migrate/20190123145628_create_podcasts.rb [hunk 1]
```diff
@@ -7,5 +7,6 @@ def change
       t.string :availability, null:false
       t.timestamps
     end
+
```
Comment (CassandraNewell) [inline db/migrate/20190123145628_create_podcasts.rb]: Watch for unnecessary empty lines

File: app/controllers/api/v1/podcasts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Api::V1::PodcastsController < ApplicationController
```
Comment (CassandraNewell) [inline app/controllers/api/v1/podcasts_controller.rb]: You need to add a test for this API endpoint

File: app/javascript/react/containers/PodcastIndexContainer.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  componentDidMount() {
+    fetch("/api/v1/podcasts.json")
+      .then((response) => response.json())
+      .then((responseData) => {
+        this.setState({podcasts: responseData})
+      })
+      .catch((error) => {
+        throw error
+      })
+  }
```
Comment (nwalberts) [inline app/javascript/react/containers/PodcastIndexContainer.js]: please add the complete set of standard error handling here, per the example in the original  fetch reading

File: app/javascript/react/containers/PodcastIndexContainer.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  componentDidMount() {
+    fetch("/api/v1/podcasts.json")
+      .then((response) => response.json())
+      .then((responseData) => {
+        this.setState({podcasts: responseData})
+      })
+      .catch((error) => {
+        throw error
+      })
```
Comment (nwalberts) [inline app/javascript/react/containers/PodcastIndexContainer.js]: please add the complete set of standard error handling here, per the example in the original fetch reading

File: app/javascript/react/containers/PodcastShowContainer.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  componentDidMount() {
+    fetch(`/api/v1/podcasts/${this.props.params.id}`)
+      .then((response) => response.json())
+      .then((responseData) => {
+        this.setState({ podcast: responseData })
+      })
+      .catch((error) => {
+        throw error
+      })
+  }
```
Comment (nwalberts) [inline app/javascript/react/containers/PodcastShowContainer.js]: Just need a bit more for handling the response in error handling here

File: spec/javascript/containers/PodcastShowContainer.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      expect(wrapper.state()).toEqual({podcast: podcast})
+      done()
+    }, 0)
+  })
+
+  it('should render show page contents after api call', (done) => {
+    setTimeout(() => {
+      let expectIncludes = (wrapper, value) => {
+        expect(wrapper.text().includes(value)).toEqual(true)
+      }
```
Comment (nwalberts) [inline spec/javascript/containers/PodcastShowContainer.js]: if you want to use this wonderful helper method, would you mind defining it in your test helper to be shared across all of your tests? 

Or even better, you could make a expectIncludesHelper.js file that is then required into your test helper. That would be the best separation of concerns

**Pull request merged**